### ISSUE TITLE
Announcement: friendly mantainance fork at https://github.com/fishbotics/urchin/

### ISSUE THREAD

**Issue by traversaro** [mmatl/urdfpy#31]
Hello everyone! I am helping mantain a library (https://github.com/icub-tech-iit/urdf-modifiers) that depends on urdfpy, and I am the mantainer of the urdfpy recipe on conda-forge. Unfortunatly, it is currently difficult to urdfpy with pip due to incompatibility with recent versions of Python and numpy (see https://github.com/mmatl/urdfpy/pull/23 and https://github.com/mmatl/urdfpy/pull/24). 

~Due to the lack of response from urdfpy mantainers, I created a friendly mantainance fork of this library at https://github.com/ami-iit/murdfpy/, with the latest fixes from @johannespitz and @fishbotics integrated (thanks a lot!), and I released it and made it available on PyPI: https://pypi.org/project/murdfpy/~ Actually, after opening this issue I switched to use/help  https://github.com/fishbotics/urchin/ .

The slight change of name was made to be able to upload the package to PyPI, but the API is exactly the same, to use you just need to change `urdfpy` to `urchin`, i.e. `from urdfpy import URDF` to `from urchinimport URDF`. 

If the `urdfpy` mantainers come back to mantain the library, I will be happy to take the friendly fork down, but as long as there is no reply from them, feel free to use it.

**Comment by traversaro** [comment]
fyi to possible interested users: @adamconkey @thesofakillers

**Comment by traversaro** [comment]
Exactly the second after opening this issue, I noticed that another similar fork exists at https://github.com/fishbotics/urchin/ . @fishbotics I guess the spirit of your fork is the same, right? So probably I can bring down murdfpy and just point users to urchin ?

**Comment by fishbotics** [comment]
Yes! It seems that @mmatl is no longer maintaining this package, so I forked it awhile ago to make urchin. So far, the changes have been fairly minimal (unpinned dependencies, the ability to load the kinematics without loading the meshes), but I am open to discussing future directions (perhaps in that repo instead).

@traversaro, I would be happy to make you a co-maintainer of the fork if you like. I imagine having a few people helping out will keep it from meeting a similar fate in the future.

FWIW, urdfpy (and thus urchin) depends on @mmatl's other popular package PyRender, which also appears to be unmaintained and has stale dependencies. I did the same thing with that package to make Pyribbit (https://github.com/fishbotics/pyribbit).

**Comment by adamconkey** [comment]
Thanks @fishbotics and @traversaro, this is helpful to know the lay of the land of these packages. What do you suppose would be prospect of @mmatl adding maintainers to _this_ package? I always find it a little confusing when unmaintained packages start getting passed around under different names.

That being said, I'll follow along on whatever fork you guys decide to maintain. I primarily use this package for lightweight visualization/debugging without having to fire up ROS infrastructure. If there is well-defined dev that could make that task easier and more informative, I'd be happy to try to help in my spare time.

**Comment by traversaro** [comment]
> @traversaro, I would be happy to make you a co-maintainer of the fork if you like. I imagine having a few people helping out will keep it from meeting a similar fate in the future.

Great, I would be happy to be a co-maintainer. In my fork I also added GitHub Actions support for running the tests and automatically adding wheels PyPI for any new tag (see https://github.com/ami-iit/murdfpy/blob/0.0.23/.github/workflows/python-package.yml and https://github.com/ami-iit/murdfpy/blob/0.0.23/.github/workflows/publish-to-test-pypi.yml), if you like I can submit them as PR to pyribbit and urchin.

**Comment by traversaro** [comment]
In the meanwhile, I will redirect users from murdfpy to  urchin.

**Comment by traversaro** [comment]
I renamed the issue for visibility.

**Comment by traversaro** [comment]
@fishbotics can you enable issues in https://github.com/fishbotics/urchin ? I wanted to open a discussion related to urchin, but I could not (I guess it was disabled when it was just a fork of urdfpy). Thanks!

**Comment by fishbotics** [comment]
@traversaro : added issues and made you a co-maintainer

**Comment by traversaro** [comment]
> @traversaro : added issues and made you a co-maintainer

Great, thanks!
